src/main/java/org/jenkinsci/plugins/ghprb/extensions/status/GhprbSimpleStatus.java

```diff
@@ -203,6 +216,10 @@ public void onBuildComplete(AbstractBuild<?, ?> build,
             }
         }
 
+        if((showMatrixStatus && build.getProject() instanceof MatrixProject)){
```

Comment:
From what I observed, showMatrixStatus will be the same for both the parent matrix job and the child job. So if I were to use !showMatrixStatus, it would not post commit status when we expect the default behaviour.